keptn/keptn
# feat: Signed container images

### This PR
- adds functionality to sign all released container images through sigstore/cosign
- images are signed using a a ephemeral keypairs and signatures provided by the [fulcio root CA](https://github.com/sigstore/fulcio) using keyless signing ([more info here](https://github.com/sigstore/cosign/blob/main/KEYLESS.md))
- only release and pre-release images are signed
- container image signatures are stored in the [rekor open transparency log](https://github.com/sigstore/rekor) for users to verify against

CI run with keyless signing and verification enabled: https://github.com/keptn/keptn/actions/runs/2926273519

#### How to test
- Install cosign (https://github.com/sigstore/cosign#installation)
- Run the following command to verify a test image that I signed in the CI pipeline with the following command: `COSIGN_EXPERIMENTAL=1 cosign verify keptndev/api:0.19.0-dev-PR-8740.202208251147`

Fixes #7164
